File: java/src/main/java/org/xllgl2017/Method.java
package org.xllgl2017;

public enum Method {
    GET(0),
    POST(1),
    PUT(2),
    HEAD(3),
    DELETE(4),
    OPTIONS(5),
    TRACE(6),
    CONNECT(7),
    PATCH(8);

    final int value;

    Method(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }

}

File: go/reqrio/method.go
package reqrio

import "errors"

type Method int

const (
	GET Method = iota
	POST
	PUT
	HEAD
	DELETE
	OPTIONS
	TRACE
	PATCH
	QUERY
)

func ParseStringMethod(method string) (Method, error) {
	switch method {
	case "GET":
		return GET, nil
	case "POST":
		return POST, nil
	case "PUT":
		return PUT, nil
	case "HEAD":
		return HEAD, nil
	case "DELETE":
		return DELETE, nil
	case "OPTIONS":
		return OPTIONS, nil
	case "TRACE":
		return TRACE, nil
	case "PATCH":
		return PATCH, nil
	case "QUERY":
		return QUERY, nil
	default:
		return GET, errors.New("invalid method")
	}
}
